src/components/Exchange/PositionShare.js

```diff
@@ -53,7 +53,11 @@ function PositionShare({ setIsPositionShareModalOpen, isPositionShareModalOpen,
     (async function () {
       const element = positionRef.current;
       if (element && userAffiliateCode.success && sharePositionBgImg && positionToShare) {
-        const image = await toJpeg(element, config);
+        // We have to call the toJpeg function multiple times to make sure the canvas renders all the elements like background image
+        // @refer https://github.com/tsayen/dom-to-image/issues/343#issuecomment-652831863
+        const image = await toJpeg(element, config)
+          .then(async () => await toJpeg(element, config))
```

Comment:
it's nesessary to add async / await here?

if toJpeg returns a promise, you can just write:
.then(() => toJpeg())
.then(() => toJpeg())

Reply:
I tried with this but the canvas doesn't load the background image on iOS (safari). It works okay with download but with uploading the image it doesn't. With async await it does. I will try and find a better way going forward.